### ISSUE TITLE
internal compiler error: librustc_codegen_llvm/mir/operand.rs:165: not immediate

### ISSUE THREAD

**Issue by ojii** [rust-lang/rust#51431]
Running the following code triggers an internal compiler error and the error message says that it's a bug and I should report it, so I'm doing that now: https://play.rust-lang.org/?gist=d30fc571aba5a2b7276cee44445f2a5e&version=nightly&mode=debug

**Comment by hellow554** [comment]
MCVE: 
```
#![feature(asm)]

fn main() {
    unsafe {
        asm! {"mov $0,$1"::"0"("bx"),"1"(0x00)}
    }
}
```

**Comment by hellow554** [comment]
Note, that this does not ICE anymore on nightly, instead it gives me:

```none
Compiling playground v0.0.1 (/playground)
error[E0669]: invalid value for constraint in inline assembly
  --> src/main.rs:15:11
   |
15 |     apm!(("al", 0x01), ("bx", 0x00));
   |           ^^^^

error[E0669]: invalid value for constraint in inline assembly
  --> src/main.rs:15:25
   |
15 |     apm!(("al", 0x01), ("bx", 0x00));
   |                         ^^^^

error: <inline asm>:1:11: error: unexpected token in argument list
        mov ah,53h
                 ^

  --> src/main.rs:5:13
   |
5  |             asm! {"mov ah,53h"};
   |             ^^^^^^^^^^^^^^^^^^^^
...
15 |     apm!(("al", 0x01), ("bx", 0x00));
   |     --------------------------------- in this macro invocation

error: <inline asm>:1:8: error: unexpected token in argument list
        int 15h
              ^

  --> src/main.rs:9:13
   |
9  |             asm! {"int 15h"};
   |             ^^^^^^^^^^^^^^^^^
...
15 |     apm!(("al", 0x01), ("bx", 0x00));
   |     --------------------------------- in this macro invocation

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0669`.
error: Could not compile `playground`.

To learn more, run the command again with --verbose.
```

Can this be closed?
